### PR TITLE
Bug 1408067 - Add Send to Device entry point to the Page Action Menu.

### PR DESCRIPTION
**Pull request by q2r5**

### CODE REVIEW

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 1]
```diff
@@ -1302,6 +1319,10 @@ extension BrowserViewController: URLBarDelegate {
         let actionMenuPresenter: (URL, Tab, UIView, UIPopoverArrowDirection) -> Void  = { (url, tab, view, _) in
             self.presentActivityViewController(url, tab: tab, sourceView: view, sourceRect: view.bounds, arrowDirection: .up)
         }
+
+        let sendToDeviceAction = {
```
Comment (farhanpatel) [inline Client/Frontend/Browser/BrowserViewController.swift]: I think can actually move the entire `presentSendToDevice` into PhotonActionSheetProtocol. You dont need to create and pass this closure in.

File: Client/Frontend/Widgets/PhotonActionSheetProtocol.swift [hunk 2]
```diff
@@ -177,6 +178,10 @@ extension PhotonActionSheetProtocol {
                 _ = self.profile.history.addPinnedTopSite(site).value
             }
         }
+
+        let sendToDevice = PhotonActionSheetItem(title: Strings.AppMenuSendToDeviceTitleString, iconString: "menu-Send-to-Device") { action in
+            presentSendTo()
```
Comment (farhanpatel) [inline Client/Frontend/Widgets/PhotonActionSheetProtocol.swift]: all the code in the BVC can move here 😄 
`presentableVC` is actually the BVC. so you can simply do `presentableVC.present` 

you need to double check if presentableVC conforms to the delegate protocols used by the two VCs 
`           guard let bvc = presentableVC as? InstructionsViewControllerDelegate & ClientPickerViewControllerDelegate else {
                return
            }
`

**Comment by q2r5** [comment]
@farhanpatel Updated to implement the requested changes.  r?

File: Client/Frontend/Widgets/PhotonActionSheetProtocol.swift [hunk 3]
```diff
@@ -177,6 +177,23 @@ extension PhotonActionSheetProtocol {
...
             }
         }
+
+        let sendToDevice = PhotonActionSheetItem(title: Strings.AppMenuSendToDeviceTitleString, iconString: "menu-Send-to-Device") { action in
+            guard let bvc = presentableVC as? PresentableVC & InstructionsViewControllerDelegate & ClientViewControllerDelegate else { return }
+            if !self.profile.hasAccount() {
+                let instructionsViewController = InstructionsViewController()
+                instructionsViewController.delegate = bvc
+                let navigationController = UINavigationController(rootViewController: instructionsViewController)
+                bvc.present(navigationController, animated: true, completion: nil)
```
Comment (garvankeeley) [inline Client/Frontend/Widgets/PhotonActionSheetProtocol.swift]: Should there be a return here?

File: Client/Frontend/Widgets/PhotonActionSheetProtocol.swift [hunk 4]
```diff
@@ -177,6 +177,23 @@ extension PhotonActionSheetProtocol {
                 _ = self.profile.history.addPinnedTopSite(site).value
             }
         }
+
+        let sendToDevice = PhotonActionSheetItem(title: Strings.AppMenuSendToDeviceTitleString, iconString: "menu-Send-to-Device") { action in
+            guard let bvc = presentableVC as? PresentableVC & InstructionsViewControllerDelegate & ClientViewControllerDelegate else { return }
```
Comment (farhanpatel) [inline Client/Frontend/Widgets/PhotonActionSheetProtocol.swift]: this should be `ClientPickerViewControllerDelegate` ?

File: Client/Frontend/Widgets/PhotonActionSheetProtocol.swift (hunk 3 above)
Comment (farhanpatel) [reply Client/Frontend/Widgets/PhotonActionSheetProtocol.swift]: Yea makes sense. Lets put a return here so a `ClientPickerViewController` is not also created.

File: Client/Frontend/Widgets/PhotonActionSheetProtocol.swift [hunk 5]
```diff
@@ -195,7 +212,7 @@ extension PhotonActionSheetProtocol {
             topActions.append(addReadingList)
         }
         
-        return [topActions, [copyURL, findInPageAction, toggleDesktopSite, pinToTopSites], [share]]
+        return [topActions, [copyURL, findInPageAction, toggleDesktopSite, pinToTopSites, sendToDevice], [share]]
```
Comment (farhanpatel) [inline Client/Frontend/Widgets/PhotonActionSheetProtocol.swift]: This feature has a new string so it wont be able to land in 10. We'll need to add it behind a feature flag. This should be pretty simple. 
We create a new entry in `AppConstants.swift` and set the feature disabled in beta/release and then here you can just do 
```
        var middleActions = [copyURL, findInPageAction, toggleDesktopSite, pinToTopSites]
        if MOZ_SENDTAB_IN_PAGE_ACTIONS {
            middleActions.append(sendToDevice)
        }
        
        return [topActions, middleActions, [share]]
```

Comment (farhanpatel) [reply Client/Frontend/Widgets/PhotonActionSheetProtocol.swift]: We might be able to rework a few strings to have this land in 10. I'll have to talk to Stefan about this.

**Comment by q2r5** [comment]
@farhanpatel Changes implemented. Looking good now?

**Comment by farhanpatel** [comment]
Thank you. This is wonderful 🥇

**Pull request merged**